### PR TITLE
Add Quality Control (QC) testing

### PR DESCRIPTION
**Pull request by darincomeau**
Adds a QC testing script adapted for MPAS-Seaice from CICE implementation: https://github.com/CICE-Consortium/CICE/blob/main/configuration/scripts/tests/QC/cice.t-test.py

Usage instructions in `README.md`.

Still a work in progress; the QC test itself runs, but plotting results does not yet work.

### CODE REVIEW

File: components/mpas-seaice/testing_and_setup/qc_testing/mpas-seaice.t-test.py [hunk 1]
```diff
@@ -0,0 +1,818 @@
...
+        '''
+        Calculate the difference and mask the points where the the difference at
+        every timestep is 0, or the sea ice thickness for every timestep is < 0.01 meters
+        for data_a or data_b
+        '''
+        logging.info("Diffing data")
+        data_d = data_a - data_b
+        mask_d = np.logical_or(\
+                      np.logical_or(\
+                           np.all(np.equal(data_d, 0.), axis=0), np.all(data_a < 0.01, axis=0))\
```
Comment (darincomeau) [inline components/mpas-seaice/testing_and_setup/qc_testing/mpas-seaice.t-test.py:127]: Noting the threshold of 0.01

**Comment by darincomeau** [comment]
Added reviewers for visibility

**Comment by darincomeau** [comment]
As I mentioned in today's meeting, this is currently doing testing with the `iceVolume` variable, whereas the CICE implementation used ice thickness. I grepped through the code and couldn't find an `iceThickness` variable in MPAS-Seaice I could use here. @akturner could you point me to what you were referring to?

**Comment by proteanplanet** [comment]
@darincomeau Do you need a review on this yet?

**Comment by eclare108213** [comment]
Based on the instructions here, I developed a [workflow](https://acme-climate.atlassian.net/wiki/spaces/ICE/pages/3450339435/Project+Workflow) using the new E3SM_Polar_Developer.sh script to run the CICE-QC tests, and then produce comparison plots using MPAS-Analysis.  The tests are using 'actual' thickness now rather than volume, and they seems to work well.  The plotting capability will be useful when we get CICE-QC 'fail' results, but for now the MPAS-Analysis plots provide some idea of where the issues are.

**Comment by darincomeau** [comment]
Closing, PR on main repo: https://github.com/E3SM-Project/E3SM/pull/5969

**Pull request closed without merging**